app/controllers/users_controller.rb

```diff
@@ -2,6 +2,7 @@ class UsersController < ApplicationController
   def homepage
     if user_signed_in?
```

Comment:
This validation should be placed in a `before_filter` callback. One of the uses of `before_filter` is control the execution chain